include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -191,6 +192,329 @@ AffineConstraints<number>::is_consistent_in_parallel(
+                                     &requests.back());
+          AssertThrowMPI(ierr);
+        }
+
+      // ... receive data
+      std::set<unsigned int> ranks;
+
+      for (const unsigned int rank : locally_relevant_dofs_owners)
+        if (rank != my_rank)
+          ranks.insert(rank);
+
+      for (unsigned int counter = 0; counter < ranks.size(); ++counter)
+        {
+          MPI_Status status;
+          int ierr = MPI_Probe(MPI_ANY_SOURCE, tag, mpi_communicator, &status);
+          AssertThrowMPI(ierr);
+
+          int message_length;
+          ierr = MPI_Get_count(&status, MPI_CHAR, &message_length);
+          AssertThrowMPI(ierr);
+
+          std::vector<char> buffer(message_length);
+
+          ierr = MPI_Recv(buffer.data(),
+                          buffer.size(),
+                          MPI_CHAR,
+                          status.MPI_SOURCE,
+                          tag,
+                          mpi_communicator,
+                          MPI_STATUS_IGNORE);
+          AssertThrowMPI(ierr);
+
+          const auto received_locally_relevant_constrain =
+            Utilities::unpack<std::vector<ConstraintType>>(buffer, false);
+
+          for (const auto &locally_relevant_constrain :
+               received_locally_relevant_constrain)
+            locally_relevant_constrains.push_back(locally_relevant_constrain);
+        }
+
+      const int ierr =
+        MPI_Waitall(requests.size(), requests.data(), MPI_STATUSES_IGNORE);
+      AssertThrowMPI(ierr);
+    }
+
+    // step 4: sort constraints and make unique
+    std::sort(locally_relevant_constrains.begin(),
+              locally_relevant_constrains.end(),
+              [](const auto &a, const auto &b) {
+                return std::get<0>(a) < std::get<0>(b);
+              });
+    locally_relevant_constrains.erase(
+      std::unique(locally_relevant_constrains.begin(),
+                  locally_relevant_constrains.end(),
+                  [](const auto &a, const auto &b) {
+                    return std::get<0>(a) == std::get<0>(b);
+                  }),
+      locally_relevant_constrains.end());
+
+    return locally_relevant_constrains;
+  };
+
+  // 1) get all locally relevant constraints ("temporal constraint matrix")
+  const auto temporal_constraint_matrix = compute_locally_relevant_constrains();
+
+  // 2) clear the content of this constraint matrix
+  lines.clear();
+  lines_cache.clear();
+
+  // 3) refill this constraint matrix
+  for (const auto &line : temporal_constraint_matrix)
+    {
+      const types::global_dof_index index = std::get<0>(line);
+
+      // ... line
+      this->add_line(index);
+
+      // ... inhomogeneity
+      if (std::get<1>(line) != number())
+        this->set_inhomogeneity(index, std::get<1>(line));
+
+      // ... entries
+      if (std::get<2>(line).size() > 0)
+        for (const auto &j : std::get<2>(line))
+          this->add_entry(index, j.first, j.second);
+    }
+#endif
```

Comment:
Does it make sense to run is_consistent in debug mode here?

Reply:
Added.